src/picasso-definition/components/mini-chart/navigation-rect.js

```diff
@@ -0,0 +1,101 @@
+import KEYS from '../../../constants/keys';
+import NUMBERS from '../../../constants/numbers';
+
+export default function createMiniChartNavigationRect({ models, flags }) {
+  const ratio = NUMBERS.MINI_CHART.RATIO;
+  const padding = NUMBERS.MINI_CHART.PADDING; // Padding from the bottom right corner
+
+  // Home state data view (corresponding the rect of the mini chart)
+  let xAxisMin0;
+  let xAxisMax0;
+  let yAxisMin0;
+  let yAxisMax0;
+
+  // Current data view (corresponding to the rect of the navigation window)
+  let xAxisMin;
+  let xAxisMax;
+  let yAxisMin;
+  let yAxisMax;
+
+  // Width and height of the point component
+  let width;
+  let height;
+
+  // Width and height of the navigation window
+  let navWidth;
+  let navHeight;
+
+  // The top left corner of the navigation window, relative to the mini chart
+  let navX;
+  let navY;
+
+  const { chartModel, layoutService } = models;
+  const viewHandler = chartModel.query.getViewHandler();
+
+  return {
+    key: KEYS.COMPONENT.MINI_CHART_NAVIGATION,
+    type: 'custom-rect',
+    style: { borderColor: 'red', background: 'rgba(200, 200, 200, 0.3)' },
+    rect: {
```

Comment:
Put `rect` and other settings that are specific to the `custom-rect` under a `settings` object.